pytorch_lightning/utilities/fetching.py

```diff
@@ -169,40 +198,207 @@ def reset(self) -> None:
         self.fetched: int = 0
         self.done: bool = False
 
+    def teardown(self) -> None:
+        self.reset()
+
 
 class DataFetcher(AbstractDataFetcher):
 
     """
     This class is used to control batch fetching flow.
+    By default, the ``fetching_function`` will pre-fetch a batch in advance to detect the end of the iteration.
+
+    Args:
+        prefetch_batches: Number of batches to be pre-fetched. Lightning will pre-fetch
+            at least 1 batch for tracking the latest batch.
+        store_on_gpu: Whether to store the pre-fetched batches on device.
     """
 
+    def __init__(
+        self,
+        prefetch_batches: int = 0,
+        store_on_gpu: bool = False,
```

Comment:
This mainly is since we want to overlay the host-to-gpu copy here which requires some additional cuda logic (mainly streams).

But I agree, what about other accelerators?

Reply:
changed to `store_on_device`.